qa/tasks/cephfs/test_scrub_checks.py

```diff
@@ -362,6 +369,70 @@ def test_scrub_repair(self):
         # fragstat should be fixed
         self.mount_a.run_shell(["rmdir", test_dir])
 
+    def test_scrub_merge_dirfrags(self):
+        """
+        That a directory is merged during scrub.
+        """
+
+        abs_test_path = "/testdir"
+        split_size = 20
+        merge_size = 5
+        split_bits = 1
+
+        self.fs.mds_asok(['config', 'set', 'mds_bal_split_size', str(split_size)])
+        self.fs.mds_asok(['config', 'set', 'mds_bal_merge_size', str(merge_size)])
+        self.fs.mds_asok(['config', 'set', 'mds_bal_split_bits', str(split_bits)])
+
+        self.mount_a.create_n_files("%s/file".format(abs_test_path), split_size * 2)
+        
+        self.fs.mds_asok(['flush', 'journal'])
+        self.wait_until_true(
+            lambda: self.get_splits() == 1,
+            timeout=30
+        )
+
+        split_size = 100
+        merge_size = 30
+
+        self.fs.mds_asok(['config', 'set', 'mds_bal_split_size', str(split_size)])
+        self.fs.mds_asok(['config', 'set', 'mds_bal_merge_size', str(merge_size)])
+        self.assertEqual(self.get_merges(), 0)
+        out_json = self.fs.run_scrub(["start", abs_test_path, "recursive"])
+
+        self.wait_until_true(
+            lambda: self.get_merges() == 1,
+            timeout=30
+        )
+
+        self.mount_a.run_shell(["rm", "-f", run.Raw("{path}/file*".format(path=abs_test_path))])
```

Comment:
no need to do cleanup when a test ends, the entire file system is deleted between tests

Reply:
Done.